roles/docker_node/tasks/docker-ubuntu.yml

```diff
@@ -0,0 +1,74 @@
+- name: Install prerequisites
+  apt:
+    name: "{{ ubuntu_prerequisites }}"
+    state: present
+    update_cache: yes
+  become: true
+
+- name: Create keyrings directory
+  file:
+    path: /etc/apt/keyrings
+    state: directory
+    mode: "0755"
+  become: true
+
+- name: Add Docker GPG key
+  shell: curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+  args:
+    creates: /etc/apt/keyrings/docker.asc
+  become: true
+
+- name: Set permissions on GPG key
+  file:
+    path: /etc/apt/keyrings/docker.asc
+    mode: "0644"
+  become: true
+
+- name: Add Docker repository
+  apt_repository:
+    repo: "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable"
```

Comment:
We should probably make the system architecture configurable too, we are using amd64 but others may be using something else